src/utils/breaches.ts

```diff
@@ -0,0 +1,212 @@
+  for (const email of monitoredEmails) {
+    if (email.verified) {
+      verifiedEmails.push(
+        await bundleVerifiedEmails({
+          user,
+          email: user.primary_email,
+          recordId: email.id,
+          recordVerified: email.verified,
+          allBreaches,
+        }),
+      );
+    } else {
+      unverifiedEmails.push(email);
+    }
+  }
+
+  // get new breaches since last shown
+  for (const emailEntry of verifiedEmails) {
+    // /** @type {any[]} */
+    const newBreachesForEmail = emailEntry.breaches.filter(
+      (breach) => breach.AddedDate >= user.breaches_last_shown,
+    );
+
+    for (const newBreachForEmail of newBreachesForEmail) {
+      newBreachForEmail.NewBreach = true; // add "NewBreach" property to the new breach.
+      emailEntry.hasNewBreaches = newBreachesForEmail.length; // add the number of new breaches to the email
+    }
+  }
+
+  return { verifiedEmails, unverifiedEmails };
+}
+
+function addRecencyIndex(foundBreaches: HibpLikeDbBreach[]) {
+  // /**
+  //  * @type {any[]}
+  //  */
+  const annotatedBreaches: HibpLikeDbBreach[] = [];
+  // slice() the array to make a copy so before reversing so we don't
+  // reverse foundBreaches in-place
+  const oldestToNewestFoundBreaches = foundBreaches.slice().reverse();
+  oldestToNewestFoundBreaches.forEach((annotatingBreach, index) => {
+    const foundBreach = foundBreaches.find(
+      (foundBreach) => foundBreach.Name === annotatingBreach.Name,
+    );
+    annotatedBreaches.push(Object.assign({ recencyIndex: index }, foundBreach));
+  });
+  return annotatedBreaches.reverse();
+}
+
+type options = {
+  user: userType;
+  email: string;
+  recordId: number;
+  recordVerified: boolean;
+  allBreaches: HibpLikeDbBreach[];
+};
+async function bundleVerifiedEmails(
+  options: options,
+): Promise<BundledVerifiedEmails> {
+  const { user, email, recordId, recordVerified, allBreaches } = options;
+  const lowerCaseEmailSha = getSha1(email.toLowerCase());
+
+  // find all breaches relevant to the current email
+  const foundBreaches = await getBreachesForEmail(
+    lowerCaseEmailSha,
+    allBreaches,
+    true,
+    false,
+  );
+
+  // TODO: remove after migration MNTOR-978
+  // adding index to breaches based on recency
+  const foundBreachesWithRecency = addRecencyIndex(foundBreaches);
+
+  if (!user) {
+    const errMsg = "breachResolutionV2: subscriber cannot be undefined";
+    console.error(errMsg);
+    captureMessage(errMsg);
+
+    // @ts-ignore: function will be deprecated
+    return { verifiedEmails, unverifiedEmails };
+  }
+
+  // get v2 "breach_resolution" object
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const breachResolutionV2: any = user.breach_resolution
+    ? user.breach_resolution[email]
+      ? user.breach_resolution[email]
+      : {}
+    : [];
+
+  const useBreachId = user.breach_resolution?.useBreachId;
+
+  // Not entirely sure what this section of the code is doing, but it's not typed
```

Comment:
From git history, looks like this is backwards-compat code for https://github.com/mozilla/blurts-server/pull/2757/ - @mansaj should we keep this in place or can it be removed now? If it needs to stay I think having a comment here to explain the situation would be good.

Reply:
This recency code is no longer necessary, though we can refactor in another PR (as we talked about in the tech meeting)